### DISCUSSION TITLE
How to add custom highlighting

### DISCUSSION THREAD

**Discussion by mzfr** [adityatelange/hugo-PaperMod#119] [Q&A] [answered]
Hey,

I read issue #35 but I am unable to use highlight.js to use monokai syntax highlighting.

Can you please give some detailed steps using which I can add custom highlight.

Thanks

**Comment by pacientes** [comment]
em..

first, add your `config.yml` file in `pygmentsUseClasses: true`
and then, modify(replace, but don't change filename) `theme/hugo-PaperMod/assets/css/an-old-hope.min.css` to [this link files](https://github.com/highlightjs/highlight.js/tree/master/src/styles)

highlight sample is [here](https://highlightjs.org/static/demo/)

**Comment by mzfr** [comment]
@pacientes So I have to change the theme's original file? Is there any way of adding highlights without changing the original file? I am asking cause for the build I am using submodule + github action so modifying it on my own repo would be difficult

**Comment by pacientes** [comment]
@mzfr I haven't found a way yet.
I'm sorry T-T

**Comment by adityatelange** [comment]
> and then, modify(replace, but don't change filename) 

Any .css file is bundled, filenames can be changed no issue !
but the file may get doubled up 
Refer snippet below

https://github.com/adityatelange/hugo-PaperMod/blob/11c32302144c0f91734525bbe1f1d11ed1a16022/layouts/partials/head.html#L23-L31

**Comment by mzfr** [comment]
@adityatelange So I need to make a new css file in `/assets` directory of the theme? or my blogs root directory?

**Reply by adityatelange** [reply]
https://adityatelange.github.io/hugo-PaperMod/posts/papermod/papermod-how-to/#override-theme-template

**Comment by adityatelange** [comment]
> @pacientes So I have to change the theme's original file? Is there any way of adding highlights without changing the original file? I am asking cause for the build I am using submodule + github action so modifying it on my own repo would be difficult

when you'll put `somestyle.css` as `an-old-hope.css` , theme's `an-old-hope.css` won't be bundled but the file placed in `assets/css/an-old-hope.css` will be.
 @pacientes is correct

**Comment by mzfr** [comment]
@adityatelange I am sorry but I am still confused with the steps needed to modify the highlighting theme.

Below I am listing what I understood please correct me if I am wrong:

1) Add `pygmentsUseClasses: true`
2) Download the `.js` file from https://cdnjs.com/libraries/. Ex: for monokai(https://cdnjs.cloudflare.com/ajax/libs/highlight.js/10.4.1/styles/monokai.min.css)
3) Place it `mysite/static/css/`
4) Add a `extend_head.html` in `mysite/layout/_default/` which will include the `css` file?

Are these the right steps? If not can you please tell me the correct one

Thanks a lot for your help :)

**Comment by adityatelange** [comment, marked as answer]
1. create `layouts/partials/extend_head.html`
2. visit https://cdnjs.com/libraries/highlight.js/10.2.1
3. Select styling
![image](https://user-images.githubusercontent.com/21258296/101723629-af171200-3ad2-11eb-9359-5f035a67d1e6.png)
4. You can see all the styling down below
5. click on copy link tag
![image](https://user-images.githubusercontent.com/21258296/101723766-f56c7100-3ad2-11eb-961f-c701dd94a863.png)
6. Paste it in `layouts/partials/extend_head.html` 
![image](https://user-images.githubusercontent.com/21258296/101724045-99561c80-3ad3-11eb-96df-9fc2dff9647a.png)

7. Custom style loaded :tada:

**Comment by mzfr** [comment]
@adityatelange Thanks for the help. It is working now :)

**Comment by mfaani** [comment]
@adityatelange I can get the custom syntax highlighting applied. However the `code` css background color often makes the coloring weird. 

There's no way that that we can get a syntax and background that match?

**Reply by adityatelange** [reply]
idk man, so long not in touch.
